pgjdbc/src/test/java/org/postgresql/jdbc/PrimitiveArraySupportTest.java

```diff
@@ -0,0 +1,258 @@
+
+public class PrimitiveArraySupportTest {
+
+  public PrimitiveArraySupport.ArrayToString<long[]> longArrays = PrimitiveArraySupport.getArrayToString(new long[] {});
+  public PrimitiveArraySupport.ArrayToString<int[]> intArrays = PrimitiveArraySupport.getArrayToString(new int[] {});
+  public PrimitiveArraySupport.ArrayToString<short[]> shortArrays = PrimitiveArraySupport
+      .getArrayToString(new short[] {});
+  public PrimitiveArraySupport.ArrayToString<double[]> doubleArrays = PrimitiveArraySupport
+      .getArrayToString(new double[] {});
+  public PrimitiveArraySupport.ArrayToString<float[]> floatArrays = PrimitiveArraySupport
+      .getArrayToString(new float[] {});
+  public PrimitiveArraySupport.ArrayToString<boolean[]> booleanArrays = PrimitiveArraySupport
+      .getArrayToString(new boolean[] {});
+
+  @Test
+  public void testLongBinary() throws Exception {
+    final long[] longs = new long[84];
+    for (int i = 0; i < 84; ++i) {
+      longs[i] = i - 3;
+    }
+
+    final PgArray pgArray = new PgArray(null, Oid.INT8_ARRAY, longArrays.toBinaryRepresentation(longs));
+
+    Object arrayObj = pgArray.getArray();
+
+    assertThat(arrayObj, instanceOf(Long[].class));
+
+    final Long[] actual = (Long[]) arrayObj;
+
+    assertEquals(longs.length, actual.length);
+
+    for (int i = 0; i < longs.length; ++i) {
+      assertEquals(Long.valueOf(longs[i]), actual[i]);
+    }
+  }
+
+  @Test
+  public void testLongToString() throws Exception {
+    final long[] longs = new long[] { 12367890987L, 987664198234L, -2982470923874L };
+
+    final String arrayString = longArrays.toArrayString(',', longs);
+
+    assertEquals("{12367890987,987664198234,-2982470923874}", arrayString);
+
+    final String altArrayString = longArrays.toArrayString(';', longs);
+
+    assertEquals("{12367890987;987664198234;-2982470923874}", altArrayString);
+  }
+
+  @Test
+  public void testIntBinary() throws Exception {
+    final int[] ints = new int[13];
+    for (int i = 0; i < 13; ++i) {
+      ints[i] = i - 3;
+    }
+
+    final PgArray pgArray = new PgArray(null, Oid.INT4_ARRAY, intArrays.toBinaryRepresentation(ints));
+
+    Object arrayObj = pgArray.getArray();
+
+    assertThat(arrayObj, instanceOf(Integer[].class));
+
+    final Integer[] actual = (Integer[]) arrayObj;
+
+    assertEquals(ints.length, actual.length);
+
+    for (int i = 0; i < ints.length; ++i) {
+      assertEquals(Integer.valueOf(ints[i]), actual[i]);
+    }
+  }
+
+  @Test
+  public void testIntToString() throws Exception {
+    final int[] ints = new int[] { 12367890, 987664198, -298247092 };
+
+    final String arrayString = intArrays.toArrayString(',', ints);
+
+    assertEquals("{12367890,987664198,-298247092}", arrayString);
+
+    final String altArrayString = intArrays.toArrayString(';', ints);
+
+    assertEquals("{12367890;987664198;-298247092}", altArrayString);
+
+  }
+
+  @Test
+  public void testShorttBinary() throws Exception {
```

Comment:
fixed.

As an aside, it appears that maybe this new test class is not getting included in the code coverage report. Is there something specific which needs to be done to get this test executed for that report?